Replace inflow and outflow fields in Transaction with amountMinor

jasonpetsod opened the pull request:
Add inflow() and outflow() methods that utilize amountMinor. Change components
to use these new methods.

Fix tests to not pass both inflow and outflow fields to Transaction().

Issue: #15

Commit:
Change MutableTransactionRow to not store a Transaction in state

User inputs in a MutableTransactionRow don't necessarily map 1:1 with the
fields in a Transaction object; stop abusing Transaction objects as temporary
storage for user input. This will become especially true now that we're
starting to implement the actual data model: e.g. the 'account' and 'category'
fields will point to actual objects instead of being strings.

More immediately though, this change is required to start storing amountMinor
in Transaction instead of separate 'inflow' and 'outflow' fields.

Also stop passing 'id' to the ctor of Transaction when creating new
transactions. Population of the Transaction id is handled by addTransaction
since we use Firebase keys as the Transaction id.

Yes, this basically undoes commit 7efe39846b01c15cddb1ce98361cb1c0fe3dab63.

Issue: #15

Commit:
Replace inflow and outflow fields in Transaction with amountMinor

Add inflow() and outflow() methods that utilize amountMinor. Change components
to use these new methods.

Fix tests to not pass both inflow and outflow fields to Transaction().

Issue: #15

witriew commented on src/MutableTransactionRow.jsx:
```diff
@@ -35,8 +35,8 @@ export default class MutableTransactionRow extends React.Component {
         memo: this.props.initialTransaction.memo,
-        outflow: this.props.initialTransaction.outflow,
-        inflow: this.props.initialTransaction.inflow,
+        outflow: this.props.initialTransaction.outflow(),
```
Is it really being replaced if it's still being used?

jasonpetsod replied:
Clarified in the new commit (5119bbd1b62f4beaf3df5045c9174eb1a8255116).

witriew reviewed (commented):
Maybe this should be pushed down to the translation layer that communicates with Firebase.

Commit:
Move the logic to parse inflow and outflow to MutableTransactionRow

Change the Transaction ctor to only accept an amountMinor argument instead of
inflow and outflow. This makes MutableTransactionRow solely responsible for
parsing user input as it should be. Transaction now looks more like the data
model.

witriew commented on src/MutableTransactionRow.jsx:
```diff
@@ -22,6 +23,40 @@ export default class MutableTransactionRow extends React.Component {
 
+  static parseAmountMinor(inflowStr, outflowStr) {
+    const inflow = parseFloat(inflowStr);
+    if (inflowStr !== '' && isNaN(inflow)) {
```
isNaN(inflowStr)?

https://jsfiddle.net/d3sfjyt0/

jasonpetsod replied:
Yeah, addressed in PR #31.

witriew commented on src/Transaction.jsx:
```diff
@@ -2,20 +2,35 @@ import moment from 'moment';
     return moment(this.dateMs).utc().format('YYYY-MM-DD');
   }
+
+  inflow() {
```
Aren't inflows and outflows supposed to be Decimal types?

jasonpetsod replied:
Yup, addressed in PR #31.

Commit:
Merge branch 'master' into tx-amountminor
